build-logic/src/main/java/qsl/internal/Versions.java

```diff
@@ -0,0 +1,27 @@
+package qsl.internal;
+
+/**
+ * Constants used across the convention scripts.
+ */
+public final class Versions {
```

Comment:
why does this have to be in java, couldn't it be a .properties file or something?

Reply:
Can't be groovy since other convention scripts cant access it then. I dislike properties here since we have to load it for every single project then. Java here is the least painful and is on every project's classpath instantly.

Reply:
It's a bit unfortunate because for average/new contributors this is extremely un-intuitive and hard to find.

Reply:
Maybe make it clear in the README?